src/roiextractors/extractors/memmapextractors/memmapextractors.py

```diff
@@ -64,43 +62,80 @@ def get_num_channels(self):
+        buffer_gb: int = 0,
+    ):
         """
         Static method to write imaging.
 
         Parameters
         ----------
-        imaging: ImagingExtractor object
-            The EXTRACT segmentation object from which an EXTRACT native format
-            file has to be generated.
+        imaging: An ImagingExtractor object that inherited from MemmapImagingExtractor
         save_path: str
-            path to save the native format.
-        overwrite: bool
-            If True and save_path is existing, it is overwritten
+            path to save the native format to.
+        verbose: bool
+            Displays a progress bar.
+        buffer_gb: int
+            The size of the buffer in Gigabytes. Default of 0 forces chunks that cover only one image.
         """
         imaging = imaging_extractor
-        video_to_save = np.memmap(
+        file_size_in_bytes = Path(imaging.file_path).stat().st_size
+        available_memory_in_bytes = psutil.virtual_memory().available
+        buffer_size_in_bytes = buffer_gb * 10**9
+        if available_memory_in_bytes < buffer_size_in_bytes:
+            raise f"Not enough memory available memory {available_memory_in_bytes* 10**9} for buffer size {buffer_gb}"
+
+        num_frames = imaging.get_num_frames()
+        memmap_shape = imaging.video_structure.build_video_shape(n_frames=num_frames)
+        dtype = imaging.get_dtype()
+
+        # Load the memmap
+        video_memmap = np.memmap(
             save_path,
-            shape=(
-                imaging.get_num_frames(),
-                imaging.get_num_channels(),
-                imaging.get_image_size()[0],
-                imaging.get_image_size()[1],
-            ),
-            dtype=imaging.get_dtype(),
+            shape=memmap_shape,
+            dtype=dtype,
             mode="w+",
         )
 
-        if verbose:
-            for ch in range(imaging.get_num_channels()):
-                print(f"Saving channel {ch}")
-                for i in tqdm(range(imaging.get_num_frames())):
-                    plane = imaging.get_frames(i, channel=ch)
-                    video_to_save[ch, i] = plane
-        else:
-            for ch in range(imaging.get_num_channels()):
-                for i in range(imaging.get_num_frames()):
-                    plane = imaging.get_frames(i, channel=ch)
-                    video_to_save[ch, i] = plane
+        if file_size_in_bytes < buffer_size_in_bytes:
+            video_data_to_save = imaging.get_frames()
+            video_memmap[:] = video_data_to_save
 
-        video_to_save.flush()
+        else:
+            chunk_size_in_bytes = int(buffer_size_in_bytes)
+            type_size = np.dtype(dtype).itemsize
+
+            n_channels = imaging.get_num_channels()
+            pixels_per_frame = n_channels * np.product(imaging.get_image_size())
+            bytes_per_frame = type_size * pixels_per_frame
+            frames_per_chunk = chunk_size_in_bytes // bytes_per_frame
```

Comment:
If we are indeed changing from 'chunk' to 'buffer', let's swap these internal variable names too to avoid future dev confusion as well

Reply:
Please remind me what is the logic again?
You are using buffer to mean "the size of the thing that we load" and "chunk" to mean the minimal size that could be loaded. Is that correct?

Reply:
'Chunk' is a specific reference to a thing in a HDF5 dataset. We should probably avoid using the term in relation to binary blobs to avoid confusion between what we're talking about. The 'buffer' is the total amount of data being read into memory. Here, that means, really, the number of frames that you read and write at a single operation.